Domains/4-Application/Connectors/CivilInfrastructureFramework/CifCommon.ecschema.xml

```diff
@@ -833,60 +833,60 @@
-        <ECProperty propertyName="MeshSurfaceEntity_StartStationFormatted" typeName="string" displayLabel="Start Station" category="MeshSurfaceEntity_Component_Layer" priority="299940" kindOfQuantity="cifu:STATION"/>
-        <ECProperty propertyName="MeshSurfaceEntity_EndStationFormatted" typeName="string" displayLabel="End Station" category="MeshSurfaceEntity_Component_Layer" priority="299940" kindOfQuantity="cifu:STATION"/>
-    </ECEntityClass>
-    <ECRelationshipClass typeName="ElementOwnsAggregateMeshSurfaceEntityAspect" modifier="None" strength="embedding">
-        <BaseClass>bis:ElementOwnsUniqueAspect</BaseClass>
-        <Source multiplicity="(1..1)" roleLabel="owns" polymorphic="true">
-            <Class class="bis:GeometricElement3d"/>
-        </Source>
-        <Target multiplicity="(0..1)" roleLabel="is owned by" polymorphic="false">
-            <Class class="AggregateMeshSurfaceEntityAspect"/>
-        </Target>
-    </ECRelationshipClass>
-    <!-- Earthwork mesh e.g. CutSlope, FillSlope, Ditch and Trench -->
-    <ECEntityClass typeName="EarthworkMeshSurfaceEntityAspect">
-        <BaseClass>MeshSurfaceEntityBaseAspect</BaseClass>
-        <ECProperty propertyName="MeshSurfaceEntity_StartStationFormatted" typeName="string" displayLabel="Start Station" category="MeshSurfaceEntity_Component_Layer" priority="299940" kindOfQuantity="cifu:STATION"/>
+    </ECEntityClass>
+    <ECRelationshipClass typeName="ElementOwnsVolumeMeshSurfaceEntityAspect" modifier="None" strength="embedding">
+        <BaseClass>bis:ElementOwnsUniqueAspect</BaseClass>
+        <Source multiplicity="(1..1)" roleLabel="owns" polymorphic="true">
+            <Class class="bis:GeometricElement3d"/>
+        </Source>
+        <Target multiplicity="(0..1)" roleLabel="is owned by" polymorphic="false">
+            <Class class="VolumeMeshSurfaceEntityAspect"/>
+        </Target>
+    </ECRelationshipClass>
+    <!-- Aggregate mesh e.g. Ballast, Subballast -->
+    <ECEntityClass typeName="AggregateMeshSurfaceEntityAspect">
+        <BaseClass>MeshSurfaceEntityBaseAspect</BaseClass>
+        <ECProperty propertyName="MeshSurfaceEntity_CivilVolume" typeName="double" displayLabel="Volume" category="MeshQuantities_Civil_Quantities" priority="299900" kindOfQuantity="cifu:VOLUME"/>
+        <ECProperty propertyName="MeshSurfaceEntity_StartStationFormatted" typeName="string" displayLabel="Start Station" category="MeshSurfaceEntity_Component_Layer" priority="299940" kindOfQuantity="cifu:STATION"/>
+        <ECProperty propertyName="MeshSurfaceEntity_EndStationFormatted" typeName="string" displayLabel="End Station" category="MeshSurfaceEntity_Component_Layer" priority="299940" kindOfQuantity="cifu:STATION"/>
+    </ECEntityClass>
+    <ECRelationshipClass typeName="ElementOwnsAggregateMeshSurfaceEntityAspect" modifier="None" strength="embedding">
```

Comment:
You should be able to use a single base relationship class for all derived aspects if you have the relationship point to the abstract MeshSurfaceEntityBaseAspect class.

If you would prefer to have individual relationships for each derived aspect you can keep things as is and resolve this comment yourself so the change can merge.